scripts/components/Stateful.gd

```diff
@@ -0,0 +1,47 @@
+class_name StatefulComponent
+extends BaseComponent
+
+
+const COMPONENT_NAME := &"StatefulComponent"
+
+func get_component_name() -> StringName:
+	return COMPONENT_NAME
+
+var current_state: String = ""
+var state_to_state_map = {}
+var state_list = []
```

Comment:
these variables do basically the same thing

Reply:
I was planning on using those to transition between states automatically so a state doesn't need to explicitly say which state comes next.